elasticsearch/src/main/java/org/hibernate/search/elasticsearch/impl/ElasticsearchIndexManager.java

```diff
@@ -378,14 +380,23 @@ public void performOperations(List<LuceneWork> workList, IndexingMonitor monitor
 
 	@Override
 	public void performStreamOperation(LuceneWork singleOperation, IndexingMonitor monitor, boolean forceAsync) {
-		if ( singleOperation == FlushLuceneWork.INSTANCE ) {
+		BackendRequest<?> request = singleOperation.acceptIndexWorkVisitor( visitor, monitor );
+		if ( singleOperation instanceof FlushLuceneWork ) {
 			requestProcessor.awaitAsyncProcessingCompletion();
+			executeBackendRequest( request, true );
 		}
 		else {
-			BackendRequest<?> request = singleOperation.acceptIndexWorkVisitor( visitor, monitor );
+			executeBackendRequest( request, false );
```

Comment:
You're right, I got confused. Obviously if it's async by default, then forcing async shouldn't do anything... Sorry about that.

About `worker.execution`:

 1. It only applies to non-stream works (from what I can see in the code), so in our case it's not relevant
 2. Support is indeed missing for Elasticsearch. I added a ticket on JIRA so we can at least discuss it: https://hibernate.atlassian.net/browse/HSEARCH-2502